code/game/atoms.dm

```diff
@@ -280,22 +280,22 @@ var/list/blood_splatter_icons = list()
 
 //returns 1 if made bloody, returns 0 otherwise
 /atom/proc/add_blood(mob/living/carbon/M)
+	if(!M || !M.has_dna() || rejects_blood())
+		return 0

```

Comment:
this proc also runtimes when a bloody mob touches something and tries to transfer blood, but for whatever reason the mob's `bloody_hands_mob` var (i hate this var's existence) has not yet been set